en/docs/Administer/ProductAdministration/ManagingUsersAndRoles/ManagingUserStores/ConfigurePrimaryUserStore/configuring-a-read-write-active-directory-user-store.md

```diff
@@ -1,190 +1,572 @@
-                    <Property name="WriteGroups">true</Property>
-                    <Property name="GroupSearchBase">CN=Users,DC=WSO2,DC=Com</Property>
-                    <Property name="GroupEntryObjectClass">group</Property>
-                    <Property name="GroupNameAttribute">cn</Property>
-                    <Property name="GroupNameSearchFilter">(&amp;(objectClass=group)(cn=?))</Property>
-                    <Property name="GroupNameListFilter">(objectcategory=group)</Property>
-                    <Property name="MembershipAttribute">member</Property>
-                    <Property name="MemberOfAttribute">memberOf</Property>
-                    <Property name="BackLinksEnabled">true</Property>
-                    <Property name="Referral">follow</Property>
-                    <Property name="UsernameJavaRegEx">[a-zA-Z0-9._\-|/]{3,30}$</Property>
-                    <Property name="UsernameJavaScriptRegEx">^[\S]{3,30}$</Property>
-                    <Property name="UsernameJavaRegExViolationErrorMsg">Username pattern policy violated</Property>
-                    <Property name="PasswordJavaRegEx">^[\S]{5,30}$</Property>
-                    <Property name="PasswordJavaScriptRegEx">^[\S]{5,30}$</Property>
-                    <Property name="PasswordJavaRegExViolationErrorMsg">Password length should be within 5 to 30 characters</Property>
-                    <Property name="RolenameJavaRegEx">[a-zA-Z0-9._\-|/]{3,30}$</Property>
-                    <Property name="RolenameJavaScriptRegEx">^[\S]{3,30}$</Property>
-                    <Property name="SCIMEnabled">false</Property>
-                    <Property name="IsBulkImportSupported">true</Property>
-                    <Property name="EmptyRolesAllowed">true</Property>
-                    <Property name="PasswordHashMethod">PLAIN_TEXT</Property>
-                    <Property name="MultiAttributeSeparator">,</Property>
-                    <Property name="isADLDSRole">false</Property>
-                    <Property name="userAccountControl">512</Property>
-                    <Property name="MaxUserNameListLength">100</Property>     
-                    <Property name="MaxRoleNameListLength">100</Property>                     
-                    <Property name="kdcEnabled">false</Property>
-                    <Property name="defaultRealmName">WSO2.ORG</Property>
-                    <Property name="UserRolesCacheEnabled">true</Property>
-                    <Property name="ConnectionPoolingEnabled">false</Property>
-                    <Property name="LDAPConnectionTimeout">5000</Property>
-                    <Property name="ReadTimeout"/>
-                    <Property name="RetryAttempts"/>
-        </UserStoreManager>
+
+-   Navigate to `<PRODUCT_HOME>/repository/conf` directory to open `deployment.toml` file and do user_store_properties configurations. Following is the sample read-write active directory user store configurations:
```

Comment:
Please use <API-M_HOME> instead of <PRODUCT_HOME>

Reply:
fixed  c0c321d